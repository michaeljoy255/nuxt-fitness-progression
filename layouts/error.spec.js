import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueMeta from 'vue-meta'
import ErrorPage from './error'

// Error Page is stored in layouts by Nuxt convention
const localVue = createLocalVue()
localVue.use(VueMeta, { keyName: 'head' })

let wrapper

beforeEach(() => {
  const vuetify = new Vuetify()
  wrapper = mount(ErrorPage, {
    propsData: {
      error: null,
    },
    localVue,
    vuetify,
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Page > error', () => {
  test('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('error prop is object that defaults to null', () => {
    expect(ErrorPage.props.error).toMatchObject({
      type: Object,
      default: null,
    })
  })

  test('error prop can recieve a value', () => {
    const object = { statusCode: 500 }
    wrapper.setProps({ error: object })
    expect(wrapper.props().error).toMatchObject(object)
  })
})
