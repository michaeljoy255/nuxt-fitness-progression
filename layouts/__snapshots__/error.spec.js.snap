// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page > error matches the snapshot 1`] = `
"<div class=\\"container fill-height\\">
  <div class=\\"row\\">
    <div justify=\\"center\\" align=\\"center\\" class=\\"col\\">
      <h1>
        Sorry, an error occurred
      </h1> <a to=\\"/\\">
        Return To Home Page
      </a>
    </div>
  </div>
</div>"
`;
